File: Min_Elem/Min.py
"""
    Author : Pakshal Shashikant Jain 
    Date : 16/07/2023
    Program : Accept N numbers from user and return the Smallest number.
              Input : N : 6
                      Elements : 85 66 3 66 93 88
              Output : 3 
"""

############################################################################################################################################################

#Class Declaration
class Minimum :
    #Characteristics of Class
    i = 0;
    arr = [];
    isize = 0;

    #Constructor of Class
    def __init__(self) :
        self.i = 0;
        self.arr = [];
        self.isize = 0;
    
    #Behaviour of Class To Take Input from user
    def Accept(self) :
        No = 0;

        #Taking input from user
        print("Enter Number of Elements You Want in array");
        self.isize = int(input());

        #Inserting Elements in array 
        print("Enter Elements in array : ");
        for self.i in range(self.isize) :
            print("Enter Element Number : ",self.i + 1);
            No = int(input());

    #Behaviour of Class to find Minimum Element From Entered N Numbers
    def Min(self) :
        imin = 0;

        #Logic
        imin = self.arr[0];
        for self.i in range(self.isize) :
            if(self.arr[i] < imin) :
                imin = self.arr[i];
        
        return imin;

##########################################################################################################################################################

#Main function
def main() :
    iret = 0;

    print("Jay Ganesh.....");

    #Creating object of Minimum Class
    mobj = Minimum();

    #Call To Behaviour of Class
    mobj.Accept();
    iret = mobj.Min();

    #Printing Smallest Element From Entered N Numbers
    print("Printing Smallest Element from Entered Number : ",iret);

###############################################################################################################################################################

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main function
    main();

File: SmallestDigitInNum_RecursiveApproach/Min.java
/*
    Author : Pakshal Shashikant Jain 
    Date : 14/5/2024
    Program : Write a recursive program which accept number from user and return
              smallest digit

              Input : 87983
              Output : 3  
*/

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
import java.lang.*;
import java.util.*;

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

class MinDigit 
{
    //Characteristics of Class
    int ino;
    int imin;

    //Constructor of Class
    MinDigit(int ivalue)
    {
        this.ino = ivalue;
        this.imin = this.ino % 10;
        this.ino = this.ino / 10;
    }

    //Behaviour of Class To Find Smallest Digits
    int Min()
    {
        int Digits = 0;

        //Logic to Find
        Digits = this.ino % 10;
        if(Digits < this.imin)
        {
            this.imin = Digits;
        }
        this.ino = this.ino / 10;

        if(this.ino != 0)
        {
            //Recursive Function Call
            this.Min();
        }

        return this.imin;
    }
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Class
class Min
{
    //Main Function
    public static void main(String args[])
    {
        int ino = 0;
        int iret = 0;

        System.out.println("Jay Ganesh....");
        Scanner sobj = new Scanner(System.in);

        //Taking Input From User
        System.out.print("Enter One Number : ");
        ino = sobj.nextInt();

        //Creating object of MinDigit Class
        MinDigit mobj = new MinDigit(ino);
        //Call To Behaviour of Class
        iret = mobj.Min();

        //Printing Smallest Digit
        System.out.printf("Smallest Digit in Enter Number %d is : %d",ino,iret);
    }
}